### PR TITLE
fix: near and far alignments for orthogonal rotations

### LINKED ISSUE #910: Chart axis label alignment defaults with rotation
**Describe the bug**
The default near/far alignments for axis labels do not account for rotation of the labels and thus create some labels with strange alignment.

image

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

fix #910 

Improve axis label alignments for `near` and `far` with `0`, `90`, `-90` and `180` rotations.

Screen Recording 2020-11-20 at 03 18 PM

### Checklist

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

**Comment by nickofthyme** [comment]
jenkins retest this please

File: src/chart_types/xy_chart/renderer/canvas/axes/tick_label.ts [hunk 1]
```diff
@@ -41,6 +42,7 @@ export function renderTickLabel(ctx: CanvasRenderingContext2D, tick: AxisTick, s
     axisStyle,
     tick.position,
     position,
+    tickLabelRotation as Rotation,
```
Comment (markov00) [inline src/chart_types/xy_chart/renderer/canvas/axes/tick_label.ts:45]: `tickLabelRotation` is in degree, we don't need to cast its type to `Rotation`.
Just check, whenever need, if the rotation if 90,-90,180 or 0, legit numbers for tickLabelRotation

File: src/chart_types/xy_chart/utils/axis_utils.ts [hunk 2]
```diff
@@ -309,6 +309,7 @@ function getVerticalTextOffset(
 
 function getHorizontalAlign(
   position: Position,
+  rotation: Rotation,
```
Comment (markov00) [inline src/chart_types/xy_chart/utils/axis_utils.ts:312]: can be just `number` instead of Rotation

**Review by markov00 — APPROVED** [review]
LGTM
I've left a comment to improve a bit the typing, removing the cast that is not necessary

File: src/chart_types/xy_chart/renderer/canvas/axes/tick_label.ts (hunk 1 above)
Comment (nickofthyme) [reply src/chart_types/xy_chart/renderer/canvas/axes/tick_label.ts]: Good call, see aee4096ee9bd5e428b447a912078118053c8da87

File: src/chart_types/xy_chart/utils/axis_utils.ts (hunk 2 above)
Comment (nickofthyme) [reply src/chart_types/xy_chart/utils/axis_utils.ts]: aee4096ee9bd5e428b447a912078118053c8da87

**Pull request merged**